// google-cloud-sql/dist/lib/configurations/store.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.store = void 0;
const conf_1 = __importDefault(require("conf"));
exports.store = new conf_1.default({
    configName: 'configurations',
    projectSuffix: '',
    schema: {
        configurations: {
            type: 'array',
            default: [],
            items: {
                type: 'object',
                properties: {
                    configurationName: { type: 'string' },
                    googleCloudSqlInstance: {
                        type: 'object',
                        properties: {
                            connectionName: { type: 'string' },
                            port: { type: 'number' },
                        },
                    },
                    kubernetesContext: { type: 'string' },
                    kubernetesNamespace: { type: 'string' },
                    kubernetesServiceAccount: { type: 'string' },
                    localPort: { type: 'number' },
                },
            },
        },
    },
});
