// google-cloud-sql/dist/lib/updates.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.notifyForUpdates = void 0;
const boxen_1 = __importDefault(require("boxen"));
const update_notifier_1 = __importDefault(require("update-notifier"));
const chalk_1 = require("chalk");
const version_1 = require("./version");
const name = 'google-cloud-sql';
const oneDayMs = 24 * 60 * 60 * 1000;
const notifyForUpdates = () => {
    const { update } = (0, update_notifier_1.default)({
        pkg: {
            name,
            version: version_1.version,
        },
        updateCheckInterval: oneDayMs,
    });
    if (update && update.current !== update.latest) {
        const text = `${(0, chalk_1.yellow)(update.current)} ↦ ${(0, chalk_1.green)(update.latest)}\n\n` +
            `${(0, chalk_1.blue)('brew')} upgrade ${name}\n` +
            `${(0, chalk_1.blue)('npm')} i -g ${name}`;
        const box = (0, boxen_1.default)(text, {
            title: 'Update Available',
            titleAlignment: 'center',
            borderColor: 'yellow',
            padding: 1,
            margin: 1,
        });
        console.log(box);
    }
};
exports.notifyForUpdates = notifyForUpdates;
